### PR TITLE
fixed a few issues in the archetypes

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by martinvw** [comment]
Did you also test it? :-)

**Comment by kaikreuzer** [comment]
Almost - I implemented those tests in the Tradfri binding, where they succeed.
And I had a hard time noticing the the old test never ever had a chance to succeed as we do not register handlers as OSGi services since quite a while now...

**Comment by sjka** [comment]
The build failure is caused by 

> Connection to http://download.itemis.com refused

Ouch 😉

File: tools/archetype/binding.test/src/main/resources/archetype-resources/src/test/java/__bindingIdCamelCase__OSGiTest.java [hunk 1]
```diff
@@ -35,19 +35,26 @@
...
     }
 
     @Test
     public void creationOf${bindingIdCamelCase}Handler() {
-        ${bindingIdCamelCase}Handler handler = getService(ThingHandler.class,${bindingIdCamelCase}Handler.class);
-
-        assertThat(handler, is(nullValue()));
+        assertThat(bridge.getHandler(), is(nullValue()));
+        managedThingProvider.add(bridge);
+        waitForAssert(() -> assertThat(bridge.getHandler(), notNullValue()));
```
Comment (sjka) [inline tools/archetype/binding.test/src/main/resources/archetype-resources/src/test/java/__bindingIdCamelCase__OSGiTest.java]: you wrote `is(nullValue())` above, shouldn't it be `is(notNullValue())` here? Not sure if it actually  _is_ required, but imho it should be symmetrical at least.

Comment (kaikreuzer) [inline tools/archetype/binding.test/src/main/resources/archetype-resources/src/test/java/__bindingIdCamelCase__OSGiTest.java]: ok, updated it accordingly.

**Pull request merged**